test/controllers/PreviousYearUnderpaymentControllerSpec.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import builders.{AuthBuilder, RequestBuilder}
+import mocks.{MockPartialRetriever, MockTemplateRenderer}
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.scalatest.mock.MockitoSugar
+import org.scalatestplus.play.{OneAppPerSuite, PlaySpec}
+import play.api.i18n.I18nSupport
+import play.api.test.Helpers._
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.FakeRequest
+import uk.gov.hmrc.domain.Generator
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.frontend.auth.connectors.{AuthConnector, DelegationConnector}
+import uk.gov.hmrc.play.partials.FormPartialRetriever
+import uk.gov.hmrc.renderer.TemplateRenderer
+import uk.gov.hmrc.tai.service.{AuditService, CodingComponentService, EmploymentService, PersonService}
+import uk.gov.hmrc.tai.viewModels.PreviousYearUnderpaymentViewModel
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class UnderPaymentFromPreviousYearControllerSpec extends PlaySpec
```

Comment:
Should we add a test for what the controller returns? Or that the template is passed the `PreviousYearUnderpaymentViewModel` that we expect. Thoughts? Would this add value?